src/main.rs

```diff
@@ -23,30 +28,84 @@ macro_rules! run_action {
     }};
 }
 
+fn read_os_release<P: AsRef<Path>>(zroot: P) -> io::Result<OsRelease> {
+    let zroot = zroot.as_ref();
+    let path = &zroot.join("etc/os-release");
+    let release = OsRelease::new_from(path)?;
+    Ok(release)
+}
+
+fn get_zfs_parent(s: &str) -> String {
+    let zonename = zonename::getzonename().expect("failed to get zonename");
+    if s == "" {
+        if zonename == "global" {
+            return "zones".to_string();
```

Comment:
Obviously this is just a default so it wouldn't totally break someone like me who was building on omnios by passing `-z foo/bar`

Reply:
I agree, but if there's going to be a default at all, it's going to be wrong for *someone*. The other thing I could do here is look for a pool named `zones`, if that doesn't exist look for a pool named `rpool`.

It would also be better if this read the SMF service to find the blessed system pool, but it's so rarely not `zones` I was more focused on getting the image creation part working.

Reply:
I added a comment around this so that it's apparent that some improvement could be made here.